src/de/mod.rs

```diff
@@ -1203,6 +1204,195 @@ mod tests {
             de.read_to_end(QName(b"skip")).unwrap();
 
             assert_eq!(de.next().unwrap(), End(BytesEnd::new("root")));
+            assert_eq!(de.next().unwrap(), Eof);
+        }
+
+        /// Checks that replay replayes only part of events
+        /// Test for https://github.com/tafia/quick-xml/issues/435
+        #[test]
+        fn partial_replay() {
+            let mut de = Deserializer::from_str(
+                r#"
+                <root>
+                    <skipped-1/>
+                    <skipped-2/>
+                    <inner>
+                        <skipped-3/>
+                        <skipped-4/>
+                        <target-2/>
+                    </inner>
+                    <target-1/>
+                </root>
+                "#,
+            );
+
+            // Initial conditions - both are empty
+            assert_eq!(de.read, vec![]);
```

Comment:
Just `.is_empty()` is cleaner

Reply:
I would like to keep comparing with empty vector for two reasons:
- keep uniformity of checks in the test
- if check failed, I would see actual content of a field